src/hooks/useOpener.ts

```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react';
+
+export const useOpener = () => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleIsOpen = () => {
+    setIsOpen(prevState => !prevState);
+  };
+
+  const setEditClose = () => {
+    setIsOpen(false);
+  };
+
+  const setEditOpen = () => {
+    setIsOpen(true);
+  };
```

Comment:
in this situation select one of two ways of logic in this hook, or you have 2 separate functions for open and close or one that opens and closes. Also useOpener do not closes any edit or another modificators. Its just about open or close, so name it as setIsVisible or setIsHidden, or if for one function toggleVisibility for example. You can name it as you want, just in this style